pkg/cmd/run/rerun/rerun.go

```
@@ -37,12 +39,22 @@ func NewCmdRerun(f *cmdutil.Factory, runF func(*RerunOptions) error) *cobra.Comm
 			// support `-R, --repo` override
 			opts.BaseRepo = f.BaseRepo
 
-			if len(args) > 0 {
+			if len(args) == 0 && opts.JobID == "" {
+				if !opts.IO.CanPrompt() {
+					return cmdutil.FlagErrorf("run or job ID required when not running interactively")
+				} else {
+					opts.Prompt = true
+				}
+			} else if len(args) > 0 {
 				opts.RunID = args[0]
-			} else if !opts.IO.CanPrompt() {
-				return cmdutil.FlagErrorf("run ID required when not running interactively")
-			} else {
-				opts.Prompt = true
+			}
+
+			if opts.RunID != "" && opts.JobID != "" {
+				opts.RunID = ""
+				if opts.IO.CanPrompt() {
+					cs := opts.IO.ColorScheme()
+					fmt.Fprintf(opts.IO.ErrOut, "%s both run and job IDs specified; ignoring run ID\n", cs.WarningIcon())
```

Instead of ignoring a piece of input, should we error out in this case?

---

Sure! I'll `return cmdutil.FlagErrorf` instead. 

I took the pattern from https://github.com/cli/cli/blob/d4ead71d06e0600633968b3f32d8bcf3d1f949fa/pkg/cmd/run/view/view.go#L130-L135 - would you suggest I update it there as well? That might break a user's workflow if they've relied on it.

---

For the sake of backwards compatibility, let's not change where it was allowed before